pkg/v1/tkg/client/cluster.go

```diff
@@ -237,13 +237,14 @@ func (c *TkgClient) waitForClusterCreation(regionalClusterClient clusterclient.C
 
 	c.WaitForAutoscalerDeployment(regionalClusterClient, options.ClusterName, options.TargetNamespace)
 
-	workloadClusterClient, err := clusterclient.NewClient(workloadClusterKubeconfigPath, kubeContext, clusterclient.Options{OperationTimeout: 15 * time.Minute})
-	if err != nil {
-		return errors.Wrap(err, "unable to create workload cluster client")
-	}
-
 	if !isTKGSCluster {
 		log.Info("Waiting for addons installation...")
+
+		workloadClusterClient, err := clusterclient.NewClient(workloadClusterKubeconfigPath, kubeContext, clusterclient.Options{OperationTimeout: 15 * time.Minute})
```

Comment:
Thanks for the review! It's because the TKGS classy cluster tests will fail in that step if it is outside the `if !isTKGSCluster {}` block. Seems the guy checked in this change forgot to verify whether the E2E tests can pass... After moving it inside the `if` block, the TKGS classy cluster tests can pass as it won't go in that path.

Also the `workloadClusterClient` is only used in the `if !isTKGSCluster {}` block, so I think it is supposed to be placed inside the `if` block.

Reply:
@anujc25 / @vuil - fyi